src/supla/network/wt32_eth01.h

```diff
@@ -0,0 +1,208 @@
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+*/
+
+#ifndef wt32_eth1_h__
+#define wt32_eth1_h__
+
+#include <Arduino.h>
+
+#include <ETH.h>
+#include <WiFiClientSecure.h>
+
+#include <supla/supla_lib_config.h>
+#include <supla/network/network.h>
+
+#ifdef ETH_CLK_MODE
+#undef ETH_CLK_MODE
+#endif
+#define ETH_CLK_MODE    ETH_CLOCK_GPIO0_IN
+
+// Pin# of the enable signal for the external crystal oscillator (-1 to disable for internal APLL source)
+#define ETH_POWER_PIN   16
+
+// Type of the Ethernet PHY (LAN8720 or TLK110)
+#define ETH_TYPE        ETH_PHY_LAN8720
+
+// I²C-address of Ethernet PHY (0 or 1 for LAN8720, 31 for TLK110)
+#define ETH_ADDR        1
+
+// Pin# of the I²C clock signal for the Ethernet PHY
+#define ETH_MDC_PIN     23
+
+// Pin# of the I²C IO signal for the Ethernet PHY
+#define ETH_MDIO_PIN    18
+
+static bool eth_connected = false;
+
+// TODO: change logs to supla_log
+
+namespace Supla {
+class WT32_ETH01 : public Supla::Network {
+  public:
+    WT32_ETH01(uint8_t ethmode,
+               unsigned char *ip = nullptr)
+      : Network(ip), client(nullptr), isSecured(true) {
+      if ( ethmode == 0) {
+        ETH_ADDRESS = 0;
+      } else {
+        ETH_ADDRESS = 1;
+      }
+    }
+
+    int read(void *buf, int count) {
+      _supla_int_t size = client->available();
+      if (size > 0) {
+        if (size > count) size = count;
+        long readSize = client->read((uint8_t *)buf, size);
+#ifdef SUPLA_COMM_DEBUG
+        Serial.print(F("Received: ["));
+        for (int i = 0; i < readSize; i++) {
+          Serial.print(static_cast<unsigned char *>(buf)[i], HEX);
+          Serial.print(F(" "));
+          delay(0);
+        }
+        Serial.println(F("]"));
+#endif
+        return readSize;
+      }
+      return -1;
+    }
+
+    int write(void *buf, int count) {
+#ifdef SUPLA_COMM_DEBUG
+      Serial.print(F("Sending: ["));
+      for (int i = 0; i < count; i++) {
+        Serial.print(static_cast<unsigned char *>(buf)[i], HEX);
+        Serial.print(F(" "));
+        delay(0);
+      }
+      Serial.println(F("]"));
+#endif
+      long sendSize = client->write((const uint8_t *)buf, count);
+      return sendSize;
+    }
+
+    int connect(const char *server, int port = -1) {
+      String message;
+      if (client != NULL) {
+        client = NULL;
+      }

```

Comment:
you are creating client instance with "new" operator, but there is no delete for it. So this code will leak memory on each attempt to connect

Reply:
done